azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/MessagingFactory.java

```diff
@@ -230,6 +230,7 @@ public void onConnectionError(ErrorCondition error) {
 
         if (!this.open.isDone()) {
             this.onOpenComplete(ExceptionUtil.toException(error));
+            this.getReactor().stop();
```

Comment:
A couple of things to point out here.
(a) the OnOpenComplete internally calls the `CompletableFuture.completeExceptionally` which in turn will result in an exception thrown in the `EventHubClient.createFromConnectionString` which will go unhandled and result in an exception thrown back to the client. The `EventHubClient.createFromConnectionString` should use consider using `MessagingFactory.createFromConnectionString(connectionString.toString(), retryPolicy).whenComplete` rather than `MessagingFactory.createFromConnectionString(connectionString.toString(), retryPolicy).thenApply` which will not account for Exception scenarios.
(b) Please add some unit-tests for this change

Reply:
Thanks for the input @osgigeek.
- Yes, `completablefuture.thenApply` was intentional. We want the exceptions to flow to client.